### PR TITLE
add linear servo

### PR DESCRIPTION
**Pull request by akshatadsule**

### CODE REVIEW

File: subsystems/LinearServo.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    
+    private final Servo actuator;
+
+    /**
+     * Creates a new LinearServo
+     * @param channel the PWM port 
+     */
+    public LinearServo(int channel) {
+        actuator = new Servo(channel);
+        actuator.setBounds(2.0, 1.8, 1.5, 1.2, 1.0);
```
Comment (adias408) [inline subsystems/LinearServo.java:21]: These values are specific to the AndyMark linear actuator that we got, right? Maybe change the class name to indicate that. Or better maybe have an enum that specifies the type of linear actuator (currently only the AndyMark L16-R https://www.andymark.com/products/actuator-l16-r-50mm-stroke-150-1-6v assuming that's the one we got) and then set these values based on that to hopefully avoid screwups and provide a way to extend this in the future if we try other linear actuators

File: subsystems/LinearServo.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+     * @param distance Distance to extends the servo to. Must be between [0, 1]
+     */
+    public void setExtent(double distance) {
+        actuator.set(distance);
+    }
+
+    /**
+     * Detracts the servo to its minimum length
+     */
+    public void detract() {
```
Comment (adias408) [inline subsystems/LinearServo.java:42]: I think "retract" rather than "detract" :)

File: subsystems/LinearServo.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public void extend() {
+        actuator.setSpeed(1);
+    }
+
+    /**
+     * Extends the servo to a specified length
+     * @param distance Distance to extends the servo to. Must be between [0, 1]
+     */
+    public void setExtent(double distance) {
+        actuator.set(distance);
```
Comment (adias408) [inline subsystems/LinearServo.java:36]: should we throw an exception or have an error code of some kind if someone passes something outside that range? Or should we clamp the value to be within the range?

Is 0-1 specific to this particular linear actuator or is that somehow common?

File: subsystems/LinearServo.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    public LinearServo(int channel) {
+        actuator = new Servo(channel);
+        actuator.setBounds(2.0, 1.8, 1.5, 1.2, 1.0);
+    }
+
+    /**
+     * Extends the servo to its maximum length
+     */
+    public void extend() {
+        actuator.setSpeed(1);
```
Comment (adias408) [inline subsystems/LinearServo.java:28]: is 1 for full extend and 0 for full retract common to linear actuators or specific to this one? I also wonder why we call set() to move to a specific point (presumably including all the way in and all the way out?) and we call setSpeed() to move all the way in or out. Is there a reason?

**Review by adias408 — CHANGES_REQUESTED** [review]

File: subsystems/LinearServo.java (hunk 1 above)
Comment (AkshatAdsule) [reply subsystems/LinearServo.java]: added in [3e19ba0](https://github.com/HHS-Team670/MustangLib/pull/10/commits/3e19ba056c75e327483ceac2aaf8085f83811e55) and [3a9447f](https://github.com/HHS-Team670/MustangLib/pull/10/commits/3a9447f073054c7865c3a904e26c3d78b5305e21)

File: subsystems/LinearServo.java (hunk 2 above)
Comment (AkshatAdsule) [reply subsystems/LinearServo.java]: fixed in [fc11d88](https://github.com/HHS-Team670/MustangLib/pull/10/commits/fc11d8852ac254b05c5b443273b33a7c3c45c0d8) :)

File: subsystems/LinearServo.java (hunk 3 above)
Comment (AkshatAdsule) [reply subsystems/LinearServo.java]: added in [fc11d88](https://github.com/HHS-Team670/MustangLib/pull/10/commits/fc11d8852ac254b05c5b443273b33a7c3c45c0d8)

File: subsystems/LinearServo.java (hunk 4 above)
Comment (AkshatAdsule) [reply subsystems/LinearServo.java]: yes, it should be set. FIxed in [f148aca](https://github.com/HHS-Team670/MustangLib/pull/10/commits/f148acab8de64a42e776ea716547ad6a286e2e4f)

File: utils/servo/LinearServo.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * Retracts the servo to its minimum length
+     */
+    public void retract() {
+        actuator.set(0);
+    }  
+
+    /**
+     * Extends the servo to a specified length
+     * @param distance Distance to extends the servo to. Must be between [0, 1]
```
Comment (adias408) [inline utils/servo/LinearServo.java:41]: nit: "to extend"

**Review by adias408 — APPROVED** [review]
LGTM

File: utils/servo/ServoType.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    ANDYMARK_L16(new ServoPWMConfig(2.0, 1.8, 1.5, 1.2, 1.0));
+
+    public final ServoPWMConfig config;
+    private ServoType(ServoPWMConfig config){
+        this.config = config;
+    }
+}
+
+/**
+ * Represents constrains for PWM based servos
```
[the remark this answers is not in the document]
Comment (AkshatAdsule) [reply utils/servo/ServoType.java]: fixed: [f2f1014](https://github.com/HHS-Team670/MustangLib/pull/10/commits/f2f1014582f04cda29c455719617ecf934f803b9)

**Pull request merged**